src/main/java/org/apache/cassandra/sidecar/restore/RestoreJobUtil.java

```diff
@@ -34,23 +34,38 @@
 import org.slf4j.LoggerFactory;
 
 import com.datastax.driver.core.utils.UUIDs;
-import net.jpountz.xxhash.StreamingXXHash32;
-import net.jpountz.xxhash.XXHashFactory;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
 import org.apache.cassandra.sidecar.exceptions.RestoreJobException;
 import org.apache.cassandra.sidecar.exceptions.RestoreJobFatalException;
+import org.apache.cassandra.sidecar.utils.DigestAlgorithm;
+import org.apache.cassandra.sidecar.utils.DigestAlgorithmProvider;
 
 /**
- * Utilities that only makes sense in the context of restore jobs. Avoid using it in the other scenarios.
+ * Utilities that only makes sense in the context of restore jobs.
+ *
+ * Note: Avoid using it in the other scenarios.
+ *
  */
+@Singleton
 public class RestoreJobUtil
 {
     private static final Logger LOGGER = LoggerFactory.getLogger(RestoreJobUtil.class);
-    private RestoreJobUtil() {}
     private static final int KB_512 = 512 * 1024;
     // it is part of upload id and get validated by
     // org.apache.cassandra.sidecar.utils.SSTableUploadsPathBuilder.UPLOAD_ID_PATTERN
     private static final String RESTORE_JOB_PREFIX = "c0ffee-";
     private static final int RESTORE_JOB_PREFIX_LEN = RESTORE_JOB_PREFIX.length();
+    private static final int RESTORE_JOB_DEFAULT_HASH_SEED = 0;
+
+    private DigestAlgorithmProvider digestAlgorithmProvider;
+
+    @Inject
+    public RestoreJobUtil(@Named("xxhash32") DigestAlgorithmProvider digestAlgorithmProvider)
```

Comment:
something to think for a future improvement. I think the manifest file should have metadata including the hashing algorithm, and we should pick the correct implementation based on that metadata file